c/test/legacy/cpp/test_sbp_stdio.cc

```diff
@@ -29,7 +29,7 @@ struct SbpHeaderParams {
   uint16_t sender_id;
   uint16_t msg_type;
   uint8_t payload_len;
-  msg_obs_t payload;
+  uint8_t payload[SBP_MAX_PAYLOAD_LEN];
```

Comment:
For messages with variable length arrays the legacy API generates a struct with a zero length array at the end. Something like:
```
struct sbp_obs_t {
  struct sbp_common_header_t header;
  struct sbp_packed_obs_t obs[0];
};
```

The size of this struct doesn't include the variable length array. So in this instance `sizeof(SbpHeaderParams::payload) == 5` (the size of the header only) but when the handler is invoked a few lines down the payload length will be anything up to 255. This causes a buffer overflow.